components/board.upload/R/upload_server.R

```diff
@@ -884,14 +620,78 @@ UploadBoard <- function(id,
       return(pgx)
     })
 
+    # reset wizard upon initialization
+    observeEvent(
+      list(process_counter(),new_upload()), {
+      isolate({
+        lapply(names(uploaded), function(i) uploaded[[i]] <- NULL)
+        lapply(names(checklist), function(i) checklist[[i]] <- NULL)
+        upload_datatype(NULL)
+        upload_name(NULL)
+        upload_description(NULL)
+        upload_organism(NULL)
+        show_comparison_builder(FALSE)
+      })
+      wizardR::reset("upload_wizard")
+    })
 
+    # wizard lock/unlock logic
 
-    # create an observer that will hide tabs Upload if selected organism if null and show if the button proceed_to_upload is clicked
-    observeEvent(input$proceed_to_upload, {
-      # show tab Upload
-      shinyjs::runjs('document.querySelector(\'[data-value="Upload"]\').style.display = "";')
-      # check on upload tab
-      shinyjs::runjs('document.querySelector("a[data-value=\'Upload\']").click();')
+    # lock/unlock wizard for samples.csv
+    observeEvent(
+      list(uploaded$counts.csv, checked_counts, input$upload_wizard), {
+        req(input$upload_wizard == "Counts")
+        if (is.null(checked_counts()$status) || checked_counts()$status != "OK"){
```

Comment:
example on how to lock and unlock wizard, to make correct correct input of counts